packages/grid/x-data-grid-pro/src/hooks/features/rowReorder/useGridRowReorder.tsx

```diff
@@ -54,15 +59,18 @@ export const useGridRowReorder = (
 
   // TODO: remove sortModel check once row reorder is sorting compatible
   // remove treeDepth once row reorder is tree compatible
-  const isRowReorderDisabled = React.useMemo((): boolean => {
-    return !props.rowReordering || !!sortModel.length || treeDepth !== 1;
-  }, [props.rowReordering, sortModel, treeDepth]);
+  const isRowReorderDisabled = React.useCallback(
+    (rowParams: GridRowParams): boolean => {
+      return !rowReordering || !!sortModel.length || treeDepth !== 1 || !isRowDraggable(rowParams);
+    },
+    [rowReordering, isRowDraggable, sortModel, treeDepth],
+  );
 
   const handleDragStart = React.useCallback<GridEventListener<'rowDragStart'>>(
     (params, event) => {
       // Call the gridEditRowsStateSelector directly to avoid infnite loop
       const editRowsState = gridEditRowsStateSelector(apiRef.current.state);
-      if (isRowReorderDisabled || Object.keys(editRowsState).length !== 0) {
+      if (isRowReorderDisabled(params) || Object.keys(editRowsState).length !== 0) {
```

Comment:
Do we need this check if `rowDragStart` is already only fired if the row is draggable?

Reply:
Probably not, but you once might try and reorder rows programatically?